crates/driver/src/infra/solver/mod.rs

```diff
@@ -97,25 +100,34 @@ pub struct Config {
     pub account: ethcontract::Account,
     /// How much time to spend for each step of the solving and competition.
     pub timeouts: Timeouts,
+    /// HTTP headers that should be added to every request.
+    pub request_headers: HashMap<String, String>
 }
 
 impl Solver {
-    pub fn new(config: Config, eth: Ethereum) -> Self {
+    pub fn new(config: Config, eth: Ethereum) -> Result<Self> {
         let mut headers = reqwest::header::HeaderMap::new();
         headers.insert(
             reqwest::header::CONTENT_TYPE,
             "application/json".parse().unwrap(),
         );
         headers.insert(reqwest::header::ACCEPT, "application/json".parse().unwrap());
-        // TODO(#907) Also add an auth header
-        Self {
-            client: reqwest::ClientBuilder::new()
-                .default_headers(headers)
-                .build()
-                .unwrap(),
-            config,
-            eth,
+
+        for (key, val) in config.request_headers.iter() {
+            let header_name = HeaderName::try_from(key)?;
+            headers.insert(header_name, val.parse()?);
         }
+
+        Ok(
+            Self {
+                client: reqwest::ClientBuilder::new()
+                    .default_headers(headers)
+                    .build()
+                    .unwrap(),
```

Comment:
nit: This should also return an error now instead of panicing here.

Reply:
i'm not really sure what to make of that. Can you elaborate please and how I should go about fixing the linting issue?

Reply:
`reqwest::ClientBuilder::new().default_headers(headers).build()` returns a `Result`. If this is of type `Err` we panic (ie crash) right here due to the use of `unwrap`. However, the `new` method itself returns a `Result` as well so it feels weird to have a codepath in it which panics. Instead we should forward the `Err` as a return value of `new` (and let the caller of `new` handle it)

Concretely, you should be able to replace `.unwrap()` with a `?` (might need some error type conversion)

Reply:
Ah yea that makes sense.